server/utils/Reddit.js

```diff
@@ -0,0 +1,56 @@
+const ENDPOINT = "http://www.reddit.com/r/all/search.json";
+
+// const Reddit = {
+//   search(term, limit) {
+//     fetch(`${ENDPOINT}?q=${term}&limit=${limit}`)
+//       .then((res) => {
+//         if (res.ok) return res.json();
+//         throw Error("Request failed");
+//       })
+//       .then((res) => {
+//         console.log(res.data);
+//         return res.data.children.map((post) => ({
+//           content: post.data.url,
+//           title: post.data.title,
+//           platform: "reddit",
+//           image: post.data.thumbnail, // "self" or "default" if no thumbnail link
+//           date: new Date(post.data.created * 1000),
+//           popularity: post.data.ups,
+//         }));
+//       })
+//       .catch((err) => {
+//         console.log(err.message);
+//       });
+//   },
+// };
+
+const searchRecursive = (term, after, posts) => {
```

Comment:
It's an odd practice to take in an array as an argument and modify it as a side effect. It would be better to take posts=[] as your initial array and then return it at the end

Reply:
Thanks for this! Based on what I understood, I set a default value for posts to be [] and return it when the condition is met. I made the new changes, if you have any feedback, it'd be very valuable!